ls1intum/Artemis
# `Development`: Speed up client tests for programming exam submissions and programming exercise instructions

### Checklist
#### Client
- [x] I ran the tests locally and all tests passed.
- [x] Test pass when running locally with memory leak check.
    - Achieved by refactoring some usages of `lodash/fp`.
    - Only one use remaining in `data-table.component`.
    - When trying to refactor the last one, the number of tests failing the leak check goes up to anywhere between 60 and 150. ⟹ Separate PR in the future.

### Motivation and Context
- Local time needed to run this test ≅60s ⟹ 300ms
    - GHActions >80s ⟹ 7.6s
    - Bamboo: time no longer reported

### Steps for testing
Open a programming exercise in the code editor as instructor.
- The code editor should show all text files and folders of the repository in the overview to the left.
    - Binary files should not be shown in the overview.
    - When changing/deleting files the code editor should show that files have been changed.
    - After submitting the changes and refreshing the page the changes should still be there.
- Edit the problem statement:
    - Try to change the name of test cases referred to in the problem statement. An annotation should show up that there are issues with the problem statement. When changing other parts of the problem statement, the issue annotation in the gutter should move to the correct location.
    - When editing the problem statement, the test cases of the exercise should be listed suggestions to be inserted.


### Review Progress
#### Code Review
#### Manual Tests
